ignite/handlers/early_stopping.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+
+class EarlyStopping(object):
+    """EarlyStopping handler can be used to stop the training if no improvement after a given number of events
+
+    Args:
+        patience (int):
+            Number of events to wait if no improvement and then stop the training
+        score_function (Callable):
+            It should be a function taking a single argument, an `ignite.engines.Engine` object,
+            and return a score `float`. An improvement is considered if the score is higher.
+
+    Examples:
+    ```python
+    from ignite.engines import Engine, Events
+    from ignite.handlers import EarlyStopping
+
+    def score_function(engine):
+        val_loss = engine.state.metrics['nll']
+        return -val_loss
+
+    handler = EarlyStopping(patience=10, score_function=score_function, verbose_logger=print)
+    evaluator.add_event_handler(Events.COMPLETED, handler, trainer)
+    ```
+    """
+    def __init__(self, patience, score_function):
+        assert callable(score_function), "Argument score_function should be a function"
+        assert patience > 0, "Argument patience should be positive"
+        self.score_function = score_function
+        self.patience = patience
+        self.counter = 0
+        self.best_score = None
+        self._logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
+        self._logger.addHandler(logging.NullHandler())
+
+    def __call__(self, engine, trainer):
```

Comment:
what do you think about passing the trainer into the constructor instead, this was users don't have to remember to add the additional arg in `add_event_handler`?